Updated "getFile" to send 404 for ENOENT errors

cory-newleaf opened the pull request:
### Description

Currently, the getFile method sends a 500 error response when the requested file cannot be found, but the status code probably should be 404.

This change sets the status property of the error object to 404 when the value of the code property is equal to ENOENT.

#### Related issues

<!--
Please use the following link syntaxes:

- connect to #49 (to reference issues in the current repository)
- connect to strongloop/loopback#49 (to reference issues in another repository)
-->

- connect to #4

### Checklist

<!--
- Please mark your choice with an "x" (i.e. [x], see
https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
- PR's without test coverage will be closed.
-->

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

dhmlau commented:
@cory-newleaf , thanks for the PR.  The commit message linter is complaining the 3rd line of your commit message is longer than 72 characters.  Could you please fix it?  

@hacksparrow , the PR looks good to me.  Could you please review as well?  Thanks.

hacksparrow commented:
@cory-newleaf we'll need the corresponding tests to land this.

cory-newleaf commented:
Hey guys, sorry I've been slammed at my day job lately so I haven't had the chance to look at the pull request.

@dhmlau I've updated the commit message, and it looks like the Commit Linter test passed. 

@hacksparrow Sorry about that! I have added 2 new tests named "should not get a file from a container" to the fs.test.js and storage-service.test.js files. This was my first time coding automated tests, so you may want to take a look and confirm that I coded them properly.

**Additional Note:** the travis build is failing due to Node v4.9.1 not supporting the use of the spread operator which is used in strong-globalize v4.1.1. I believe that the recent changes to the package.json in the update to loopback v3.20.0 is responsible.

dhmlau commented:
@slnode test please
@cory-newleaf: Let me kick off CI again.  It seems to be stuck. 
Don't worry about Node 4, since it's EOL, we are dropping supporting for Node 4 as well.

cory-newleaf commented:
@dhmlau Okay sounds good!

It seems that some of the CI checks failed though :/

virkt25 commented:
Hi @cory-newleaf if you rebase the PR on master, all CI systems should pass now as I made the necessary changes to drop Node 4.

cory-newleaf commented:
Hey @virkt25 I've rebased your changes into my branch.

Looks like all checks should be green! Just waiting on the last one to complete.

virkt25 commented:
@slnode test please

virkt25 commented:
Your patch has been landed @cory-newleaf. Thanks for your contribution! You can expect a release shortly.

cory-arcticleaf commented:
@virkt25 No problem! Glad I was able to contribute!
